From 05a7256854b2a53ce00a2adf4302c7504887e425 Mon Sep 17 00:00:00 2001
From: Niklas Haas <git@haasn.xyz>
Date: Sat, 9 Jan 2021 10:06:26 +0100
Subject: [PATCH] shaders/colorspace: actually enable subgroup operations

Gigantic oops. This must have been left over from testing/benchmarking.
Now, the speedup actually *works*!
---
 src/shaders/colorspace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shaders/colorspace.c b/src/shaders/colorspace.c
index ba341ce..f6ed9e2 100644
--- a/src/shaders/colorspace.c
+++ b/src/shaders/colorspace.c
@@ -704,7 +704,7 @@ bool pl_shader_detect_peak(struct pl_shader *sh,
          log_min, sig_scale, log_scale);
 
     // Update the work group's shared atomics
-    if (gpu->caps & PL_GPU_CAP_SUBGROUPS && false) {
+    if (gpu->caps & PL_GPU_CAP_SUBGROUPS) {
         GLSL("int group_max = subgroupMax(isig_max);    \n"
              "int group_sum = subgroupAdd(isig_log);    \n"
              "if (subgroupElect()) {                    \n"
-- 
GitLab

